### PR TITLE
Generated link in backend should also give back the base url

### LINKED ISSUE #1637: 🔧 [Refactor] generate the full link in backend
## 🔧 Refactor ticket

This is also a discussion:

Generate the full link in the backend and transmit it in full to the frontend.
Currently we cannot serve proper links to Gradido-Installations which are in a subfolder `https://test.com/wallet/`. Instead the link will always point to `https://test.com/` and append the  `redeem/ABC123`.

We generate the other links like the email content etc also in the backend and configure it there - I believe we should do the same for the redeem link.

- Additional field link in GraphQL
- Configure base URL for redeem in backend
- Deliver field from backend
- Request and display link from backend instead of building it in the frontend

### PR DESCRIPTION
**Pull request by Elweyn**
## 🍰 Pullrequest
The base URL of the link should be given from the backend and not in the frontend

### Issues

- fixes #1637 

### Todo

- [X] None

### CODE REVIEW

**Review by Mogge — CHANGES_REQUESTED** [review]
Line 65 of `frontend/src/components/TransactionLinks/TransactionLink.vue`
Do you know how to grep?

File: backend/src/graphql/model/TransactionLink.ts [hunk 1]
```diff
@@ -17,6 +18,7 @@ export class TransactionLink {
     this.deletedAt = transactionLink.deletedAt
     this.redeemedAt = transactionLink.redeemedAt
     this.redeemedBy = redeemedBy
+    this.link = CONFIG.COMMUNITY_REDEEM_URL + this.code
```
Comment (ulfgebhardt) [inline backend/src/graphql/model/TransactionLink.ts:21]: regex please - like the other urls

File: frontend/src/components/ClipboardCopy.vue [hunk 2]
```diff
@@ -14,11 +14,11 @@
 export default {
   name: 'ClipboardCopy',
   props: {
-    code: { type: String, required: true },
+    link: { type: String, required: true },
   },
   data() {
     return {
-      url: `${window.location.origin}/redeem/${this.code}`,
+      url: `${this.link}`,
```
Comment (ulfgebhardt) [inline frontend/src/components/ClipboardCopy.vue:21]: rename field to link?

Comment (Elweyn) [reply frontend/src/components/ClipboardCopy.vue:21]: Yes it is not a code anymore so changed the field code to link

Comment (ulfgebhardt) [reply frontend/src/components/ClipboardCopy.vue:21]: im talking about:

```suggestion
      link: `${this.link}`,
```

wouldnt it to make sense to use the same name - is this posisble?

File: frontend/src/components/TransactionLinks/TransactionLink.vue [hunk 3]
```diff
@@ -55,14 +55,15 @@ export default {
...
+    link: { type: String, required: true },
     holdAvailableAmount: { type: String, required: true },
     id: { type: Number, required: true },
     memo: { type: String, required: true },
     validUntil: { type: String, required: true },
   },
   methods: {
     copy() {
-      const link = `${window.location.origin}/redeem/${this.code}`
+      const link = `${this.link}`
```
Comment (Mogge) [inline frontend/src/components/TransactionLinks/TransactionLink.vue:66]: Why?

File: frontend/src/components/TransactionLinks/TransactionLink.vue [hunk 4]
```diff
@@ -55,14 +55,15 @@ export default {
...
       navigator.clipboard
         .writeText(link)
```
Comment (Mogge) [inline frontend/src/components/TransactionLinks/TransactionLink.vue:68]:
```suggestion
        .writeText(this.link)
```

File: frontend/src/components/ClipboardCopy.vue [hunk 5]
```diff
@@ -14,11 +14,11 @@
 export default {
   name: 'ClipboardCopy',
   props: {
-    code: { type: String, required: true },
+    link: { type: String, required: true },
   },
   data() {
```
Comment (Mogge) [inline frontend/src/components/ClipboardCopy.vue:19]: No data needed

Comment (Mogge) [reply frontend/src/components/ClipboardCopy.vue:19]: `.writeText(this.link)` in line 27 of this version

**Review by Mogge — CHANGES_REQUESTED** [review]
Please also update the `.env.dist` in `deployment/bare_metal`

**Review by Mogge — APPROVED** [review]

**Pull request merged**